docs/new-architecture-app-intro.md

```diff
@@ -257,3 +137,371 @@ The `moduleName` has to be the same string used in the `[RCTRootView initWithBri
 // Run pod install with the flags
 RCT_NEW_ARCH_ENABLED=1 pod install
 ```
+
+## Android - Prerequisites
+
+Using the New Architecture on Android has some prerequisites that you need to meet:
+
+1. Using Gradle version >= 7.x
+2. Using Android Gradle Plugin >= 7.x (i.e. the `com.android.tools.build:gradle` dependency)
+
+If you updated to React Native 0.68+, you already meet those prerequisites. If you don't meet them, consider updating those dependencies first.
+
+## Android - React Native Gradle Plugin & Build from Source
+
+The New Architecture relies on the React Native Gradle Plugin (from the `react-native-gradle-plugin` NPM package) to build and run your project.
+
+Moreover, in this iteration of the guide we will let you build React Native from source.
+
+If you updated your project to React Native 0.68+, you probably already have this configuration set up correctly.
+
+If not make sure you edit the `android/settings.gradle` file as follows:
+
+```diff
+ rootProject.name = 'HelloWorld'
+ apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
+ include ':app'
++includeBuild('../node_modules/react-native-gradle-plugin')
+
++include(":ReactAndroid")
++project(":ReactAndroid").projectDir = file('../node_modules/react-native/ReactAndroid')
++include(":ReactAndroid:hermes-engine")
++project(":ReactAndroid:hermes-engine").projectDir = file('../node_modules/react-native/+ReactAndroid/hermes-engine')
+```
+
+Update the `android/build.gradle` file as follows:
+
+```diff
+buildscript {
+    // ...
+    dependencies {
+
+        // Add those lines
++       classpath("com.facebook.react:react-native-gradle-plugin")
++       classpath("de.undercouch:gradle-download-task:4.1.2")
+    }
+}
+```
+
+And update the `android/app/build.gradle` file (please note that this file is different than the top level `build.gradle`) as follows:
+
+```diff
+dependencies {
+  // ...
+  if (enableHermes) {
+-    def hermesPath = "../../node_modules/hermes-engine/android/";
+-    debugImplementation files(hermesPath + "hermes-debug.aar")
+-    releaseImplementation files(hermesPath + "hermes-release.aar")
++    //noinspection GradleDynamicVersion
++    implementation("com.facebook.react:hermes-engine:+") { // From node_modules
++        exclude group:'com.facebook.fbjni'
++    }
+  }
+}
+
++ configurations.all {
++     resolutionStrategy.dependencySubstitution {
++         substitute(module("com.facebook.react:react-native"))
++                 .using(project(":ReactAndroid"))
++                 .because("On New Architecture we're building React Native from source")
++         substitute(module("com.facebook.react:hermes-engine"))
++                .using(project(":ReactAndroid:hermes-engine"))
++                .because("On New Architecture we're building Hermes from source")
++     }
++ }
```

Comment:
> can't we offer a function in the gradle plugin that is reactNativeDependencies that encapsulate this code?

Short answer: we can't.

Long answer: Technically, Gradle plugins can 'expose functions', but this would make this code a little shorter.

```diff
+ configurations.all {
+     resolutionStrategy.dependencySubstitution {
+         applyFunctionFromTheGradlePlugin()
+     }
+ }
```

Which I believe has little value right now, specifically as we should potentially remove the whole `configurations.all {}` block and put it inside the Gradle Plugin as part of the `apply` phase of the plugin (i.e. the plugin will apply this configuration for this project).

So I would refrain from making a change here, that then we'll have to change again in a near future iteration.

Reply:
> can always add another configuration.all call after we setup our default settings for further customization, right?

Also this is sadly not true. If we take ownership of this and apply a `configurations.all.resolutionStrategy` directive, we're effectively imposing this change for all the users.

This means that users on Old Architecture, can't build from source anymore if they wish: https://reactnative.dev/contributing/how-to-build-from-source